File: apps/api/utils/phone.py
"""Phone normalization for DNC matching.

Returns the digit-only North-American 11-digit form (e.g. "14408405137") or
None when the input cannot be normalized. Same rules are mirrored in the
frontend at apps/web/lib/phone.ts so client- and server-side checks agree.
"""

from typing import Optional
import re


def normalize_phone(raw: Optional[str]) -> Optional[str]:
    if raw is None:
        return None
    digits = re.sub(r"\D", "", str(raw))
    if not digits:
        return None
    if len(digits) == 10:
        return "1" + digits
    if len(digits) == 11 and digits.startswith("1"):
        return digits
    return None

File: apps/web/lib/phone.ts
// Phone normalization for DNC matching. Mirrors apps/api/utils/phone.py so
// the client- and server-side checks agree. Returns the digit-only North
// American 11-digit form (e.g. "14408405137") or null when the input
// cannot be normalized.

export function normalizePhone(raw: string | null | undefined): string | null {
  if (raw === null || raw === undefined) return null;
  const digits = String(raw).replace(/\D/g, "");
  if (!digits) return null;
  if (digits.length === 10) return "1" + digits;
  if (digits.length === 11 && digits.startsWith("1")) return digits;
  return null;
}
